File: src/main/scala/com/lge/metr/LocCalculator.scala
package com.lge.metr

import scala.collection.JavaConversions.asScalaSet
import spoon.processing.AbstractProcessor
import spoon.reflect.declaration.CtClass
import spoon.reflect.code.CtBlock

class LocCalculator extends AbstractProcessor[CtClass[_]] with Naming with LocCounter {
  override def process(klass: CtClass[_]) {
    for (c <- klass.getAllMethods ++ klass.getConstructors if !c.isImplicit() && c.getBody != null) {
      println(nameFor(c) + "\t" + sloc(c) + "\t" + dloc(c))
    }
  }
}

File: src/main/java/com/lge/metr/LocCalculator.java
package com.lge.metr;

import spoon.processing.AbstractProcessor;
import spoon.reflect.declaration.CtClass;
import spoon.reflect.declaration.CtMethod;

public class LocCalculator extends AbstractProcessor<CtClass<?>> {

	@Override
	public void process(CtClass<?> kls) {
		System.out.println(kls.getQualifiedName() + "--");
		for (CtMethod<?> method : kls.getAllMethods()) {
			System.out.println("\t" + method.getSimpleName());
		}
	}

}
